frontend/src/metabase/lib/analytics.js

```diff
@@ -48,10 +61,39 @@ export const trackSchemaEvent = (schema, version, data) => {
   }
 
   if (Settings.snowplowEnabled()) {
-    trackSnowplowSchemaEvent(schema, version, data);
+    trackSnowplowSchemaEvent(schema, version, data, contextEntities);
   }
 };
 
+/**
+ * @param {string} name - Name of the action. Noun (target) + Verb in the past (action) which define the action taken - e.g. question-created, dashboard-updated, dashboard-auto-apply-filter-enabled
+ * @param {string} triggeredFrom - From where the action was taken. This can be generic like 'dashboard' or also more specific like 'dashboard_top_nav'.
+ * @param {string} targetId - (Optional) ID of the entity that the action was performed on. E.g. the ID of the question that was created in a question_created event.
+ * @param {number} durationMs - (Optional) Duration in milliseconds
+ * @param {string} result - (Optional) The outcome of the action (e.g. success, failure, …)
+ * @param {string} eventDetail - (Optional) Can be used for additional details that describe the event, e.g. the type of question that was created in a question_created event.
+ * @param {object} contextEntities - (Optional) Additional context entities (https://docs.snowplow.io/docs/understanding-your-pipeline/entities/). We send the Metabase Instance context by default with each event.
+ */
+export const trackActionEvent = (
```

Comment:
**Reviewer note:** I think it makes sense to test this with some mocked behavior. Would someone be able to point me to how we best test snowplow events?